### PR TITLE
Add Docker images for legacy Mono CI

### PR DESCRIPTION
**Pull request by directhex**
Legacy Mono has been kicked off Jenkins, so we need to use AzDO. Right now we're using existing Helix images and just have a step in our .yml to provision the extra packages we need on every build, but this is obviously a better technical solution.

The ARMv7 image is intended to allow us to run builds on our self-hosted ARM64 pool - according to the docs, correct use of LABEL allows a user-provided (distro-provided) Node binary, for cases where the VSTS-agent-provided Node binary might not work.

### CODE REVIEW

File: src/ubuntu/18.04/mono/amd64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Install additional Mono build dependencies
+RUN apt-get update \
+    && apt-get install -qq -y \
+        autoconf \
+        libtool \
+        automake \
+        build-essential \
+        gettext \
+        wget \
+        bc \
```
Comment (mthalman) [inline src/ubuntu/18.04/mono/amd64/Dockerfile:14]: Please list these in alphabetical order.  Same applies to the other Dockerfiles.

**Review by mthalman — APPROVED** [review]

File: manifest.json [hunk 2]
```diff
@@ -664,7 +664,10 @@
               "os": "linux",
               "osVersion": "bionic",
               "tags": {
-                "ubuntu-18.04-helix-arm32v7-$(System:TimeStamp)-$(System:DockerfileGitCommitSha)": {}
+                "ubuntu-18.04-helix-arm32v7-$(System:TimeStamp)-$(System:DockerfileGitCommitSha)": {},
+                "ubuntu-18.04-helix-arm32v7": {
+                    "isLocal": true
```
Comment (akoeplinger) [inline manifest.json:669]: this change seems unrelated?

File: src/ubuntu/18.04/mono/amd64/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-amd64
```
Comment (akoeplinger) [inline src/ubuntu/18.04/mono/amd64/Dockerfile:1]: I think we shouldn't derive from the Helix image since there's a lot of things we don't need from there in mono/mono, let's just copy whatever we need from there and inline here.

File: manifest.json (hunk 2 above)
Comment (directhex) [reply manifest.json:669]: It's used for the FROM calls

File: src/ubuntu/18.04/mono/amd64/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        tzdata \
+        unzip \
+        wget \
+    && rm -rf /var/lib/apt/lists/* \
+    \
+    && localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+
+ENV LANG en_US.utf8
+
+# create helixbot user and give rights to sudo without password
```
Comment (akoeplinger) [inline src/ubuntu/18.04/mono/amd64/Dockerfile:32]: let's use a different username since we won't be using this image for helix

**Review by akoeplinger — APPROVED** [review]
